packages/wallet/src/redux/protocols/funding/player-a/reducer.ts

```diff
@@ -53,7 +54,15 @@ function strategyChosen(
   if (state.type !== states.WAIT_FOR_STRATEGY_CHOICE) {
     return { protocolState: state, sharedData };
   }
-  return { protocolState: states.waitForStrategyResponse(state), sharedData };
+  const { processId, opponentAddress } = state;
+  const { strategy } = action;
+  const sentAction = strategyProposed(processId, strategy);
+  const payload = { processId, data: { sentAction } };
+  const message = messageRelayRequested(opponentAddress, payload);
```

Comment:
As we discussed I think if `handleIncomingMessage` can transform the action for Player B this could be a nice pattern to follow. I'm in favor of B.